exist-core/src/main/java/org/exist/launcher/LauncherWrapper.java

```diff
@@ -55,6 +55,8 @@ public final static void main(String[] args) {
             System.out.println("First launch: opening configuration dialog");
             ConfigurationDialog configDialog = new ConfigurationDialog(restart -> {
                 wrapper.launch();
+                // make sure the process dies when the dialog is closed
+                System.exit(0);
```

Comment:
Won't this kill both processes? You are exiting the parent, so I think it will also kill the child i.e. the launcher...

Reply:
No, a new wrapper is started in the line before, so we safely can and must stop the current one.

Reply:
@wolfgangmm I am afraid that I don't agree... at least on Unix systems. To be sure, I wrote a little test program to demonstrate the problem:

```java
public static void main(final String args[]) throws IOException, InterruptedException {
        ProcessBuilder pb = new ProcessBuilder();
        pb.command("read");  // read will run forever as it is listening to stdin
        final Process p = pb.start();

        p.waitFor(2, TimeUnit.SECONDS);

        System.out.println("sub-process `read` isAlive=" + p.isAlive());

        try {
            final Field f = p.getClass().getDeclaredField("pid");
            if (f != null) {
                f.setAccessible(true);
            }

            System.out.println("sub-process `read` pid=" + f.get(p));

        } catch (final NoSuchFieldException | IllegalAccessException e) {
            System.out.println("Could not determine pid of `read` sub-process");
        }

        System.exit(0);
        
        // when the main process exits the sub-process is also killed
        // you can check this by doing a `ps -ef | grep <pid>` where
        // <pid> is the process id reported above for the read sub-process
    }
```

This will show you that when the parent process exists the sub-process is killed.

Reply:
Actually never mind. I see my error now. In the example above the stdin is inheirted from the parent process and so when it exits, it closes stdin, which causes `read` to exit.

If I replace "read" with "/tmp/forever.sh":
```bash
#!/bin/bash
while :; do sleep 2073600; done
```

Then I see that the sub-process continues after the parent process exists. The sub-process then acquires the root process `1` as its parent.

So... LGTM sorry for the confusion.